### PR TITLE
feat(python): index config repr method

### LINKED ISSUE #1506: Feature: IndexConfig should have a __repr__ method
### SDK

Python

### Description

Trying to inspect the indices is disappointing at the moment:
```python
await table.list_indices()
```
```
[<IndexConfig at 0x149d3fb40>, <IndexConfig at 0x14d3b2f10>]
```

We should add a `__repr__` method on `IndexConfig` so we can introspect them easily here.

### PR DESCRIPTION
**Pull request by gagan-bhullar-tech**
PR fixes #1506

### CODE REVIEW

**Comment by wjones127** [comment]
Hi @gagan-bhullar-tech. Thanks for opening a PR. Unfortunately, this approach will not work. The `pyi` files are just type stubs and aren't meant to hold implementations of methods. Instead you'll need to implement a `repr` method on `IndexConfig` in the Rust files here: https://github.com/lancedb/lancedb/blob/d07d7a5980d1f064956bb417cd7aa1aba7d3f58a/python/src/index.rs#L91

Here is an example of a repr method being implemented in Rust:

https://github.com/lancedb/lancedb/blob/d07d7a5980d1f064956bb417cd7aa1aba7d3f58a/python/src/table.rs#L197-L202

Also, if you implement this, could you add a unit test that verifies the repr method shows data as expected?

**Comment by gagan-bhullar-tech** [comment]
@wjones127 thanks for pointing it out. Will do that.

**Comment by gagan-bhullar-tech** [comment]
@wjones127 can you please confirm whether test need to be written in rust or python

**Comment by gagan-bhullar-tech** [comment]
> @wjones127 can you please let me know whether test needs to be written in rust or python

Nevermind. It looks like it should be in python.

**Comment by gagan-bhullar-tech** [comment]
@wjones127 can you please review

File: python/src/index.rs [hunk 1]
```diff
@@ -98,6 +98,13 @@ pub struct IndexConfig {
     pub columns: Vec<String>,
 }
 
+#[pymethods]
+impl IndexConfig {
+    pub fn __repr__(&self) -> String {
+        format!("Index({}, {:?})", self.index_type, self.columns)
```
Comment (wjones127) [inline python/src/index.rs:104]: The first argument is probably clear, but the second argument is an arbitrary list of strings, so would be clearer what they mean if we added a name to them.
```suggestion
        format!("Index({}, columns={:?})", self.index_type, self.columns)
```

**Comment by gagan-bhullar-tech** [comment]
@wjones127 can you please review

**Review by wjones127 — APPROVED** [review]
Looks great, thanks!

**Pull request merged**